docs/source/process.mdx

```diff
@@ -340,21 +340,29 @@ You can also use [`~Dataset.map`] with indices if you set `with_indices=True`. T
 ]
 ```
 
+### Multiprocessing
+
+Multiprocessing significantly speeds up processing by parallelizing processes on the CPU. Set the `num_proc` parameter in [`~Dataset.map`] to set the number of processes to use:
+
+```py
+>>> updated_dataset = dataset.map(lambda example, idx: {"sentence2": f"{idx}: " + example["sentence2"]}, num_proc=4)
+```
+
 The [`~Dataset.map`] also works with the rank of the process if you set `with_rank=True`. This is analogous to the `with_indices` parameter. The `with_rank` parameter in the mapped function goes after the `index` one if it is already present. 
 
 ```py
 >>> from multiprocess import set_start_method
 >>> import torch
 >>> import os
 >>>
->>> set_start_method("spawn")
->>>
 >>> def gpu_computation(example, rank):
->>>     os.environ["CUDA_VISIBLE_DEVICES"] = str(rank % torch.cuda.device_count())
+>>>     torch.cuda.set_device(rank % torch.cuda.device_count())
 >>>     # Your big GPU call goes here
```

Comment:
I still would like to see a concrete example here instead of "your big GPU call goes here", cause I tried using an NLLB model with 2 GPUs to translate sentences of the [datacomp dataset](https://huggingface.co/datasets/mlfoundations/datacomp_small) in parallel and it was unclear for me how I had to do it. Should I use nn.DataParallel? Should I use .to("cuda:0") and .to("cuda:1")?

Reply:
I remember that the rank was always set to 0, so all work was done on the first GPU

Reply:
> I remember that the rank was always set to 0, so all work was done on the first GPU

This happens only if you set num_proc=1, but for multiprocessing you get multiple ranks (one per process)

Reply:
I added `model` in the example - no need to use nn.DataParallel. You just need to send the model to every GPU.

Feel free to test that the code works as expected for you !